api_server/src/request/machine_configuration.rs

```diff
@@ -42,13 +42,25 @@ impl IntoParsedRequest for VmConfig {
                 VmmAction::GetVmConfiguration(sender),
                 receiver,
             )),
-            Method::Put => {
+            Method::Patch => {
                 if self.vcpu_count.is_none()
                     && self.mem_size_mib.is_none()
                     && self.cpu_template.is_none()
                     && self.ht_enabled.is_none()
                 {
-                    return Err(String::from("Empty request."));
+                    return Err(String::from("Empty PATCH request."));
+                }
+                Ok(ParsedRequest::Sync(
+                    VmmAction::SetVmConfiguration(self, sender),
+                    receiver,
+                ))
+            }
+            Method::Put => {
+                if self.vcpu_count.is_none()
+                    || self.mem_size_mib.is_none()
+                    || self.ht_enabled.is_none()
+                {
+                    return Err(String::from("Invalid PUT request."));
```

Comment:
A better error message would be: "Missing mandatory fields."

Reply:
Thanks for the review! Error message is updated.